cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetTypeManager.java

```diff
@@ -412,10 +416,22 @@ public Boolean call() throws DatasetModuleConflictException, IOException {
           }
 
           datasetTypeMDS.deleteModule(datasetModuleId);
-          // Also delete module jar
-          Location moduleJarLocation = locationFactory.create(module.getJarLocation());
-          if (!moduleJarLocation.delete()) {
-            LOG.debug("Could not delete dataset module archive");
+          try {
+            // Also delete module jar
+            impersonator.doAs(datasetModuleId.getNamespace().toEntityId(), new Callable<Void>() {
+              @Override
+              public Void call() throws Exception {
+                Location moduleJarLocation = locationFactory.create(module.getJarLocation());
+                if (!moduleJarLocation.delete()) {
```

Comment:
Yea, its due to the FileContext object internally being created when the Location object is created. The operations on it will be with the original UGI, as long as you don't operate on it, after the delegation token expires (many hours).